Sources/SIL/SILAnalysis.swift

```diff
@@ -195,11 +196,193 @@ extension Array: AlphaConvertible where Element: AlphaConvertible {
     }
 }
 
+func substitute(_ type: Type, using s: (String) -> Type) -> Type {
```

Comment:
Perhaps, it's the time to have more strongly-typed names? Currently, we use String for a lot of things (e.g. both local and global names), so maybe it would be useful to make flavors of names apparent in the type system?

Reply:
Please, yes. I already have a few `newtype`-like wrappers in my project, and at least having type aliases would be helpful. Having said this, it's out of scope of this commit.

Reply:
Doesn't have to be done in this PR, I agree.